### DISCUSSION TITLE
One font for multiple contexts

### DISCUSSION THREAD

**Discussion by dr-dro** [mikke89/RmlUi#481] [Q&A]
I have two windows and two Rml::Context
If I use the same font or image with the same src, solid-color rectangles are displayed instead of the text in the window that was created later. I understand that this is due to internal caching. Textures are created only once, and are not available in the second window handle.
What is the correct way to duplicate resources?
In fact, I can copy the image file under a different name, but Rml::LoadFontFace("...") seems to be a context-independent method

**Comment by mikke89** [comment]
So if I understand correctly, you can't share resources between the windows. You are right that we only submit a single resource to the render interface per texture/geometry resource. Here are some suggestions to account for that in your situation:

- As a workaround, I guess you could load the font under a different name for the second window. Not ideal, but it should work similarly to your texture file approach.
- A more robust solution would be to simply submit all of the generated textures to both windows, and then simply pick the correct one when asked to render it.
- Alternatively, keep a local copy of the data, and wait to submit it to the window(s) when asked to render it.

Let me know if some of these are helpful, or if you find some other solution.
